common/oe_host_socket.h

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#ifndef _OE_HOST_SOCKET_H
+#define _OE_HOST_SOCKET_H
+
+#include <openenclave/corelibc/bits/types.h>
+#include <openenclave/internal/syscall/types.h>
+#include <stdint.h>
+
+#ifdef _WIN32
+#include <winsock2.h>
+#else
+#include <sys/socket.h>
+#endif
+
+#define GETADDRINFO_HANDLE_MAGIC 0xed11d13a
+
+typedef struct _getaddrinfo_handle
+{
+    uint32_t magic;
+    struct addrinfo* res;
+    struct addrinfo* next;
+} getaddrinfo_handle_t;
+
+OE_INLINE getaddrinfo_handle_t* _cast_getaddrinfo_handle(void* handle_)
+{
+    getaddrinfo_handle_t* handle = (getaddrinfo_handle_t*)handle_;
+
+    if (!handle || handle->magic != GETADDRINFO_HANDLE_MAGIC || !handle->res)
+        return NULL;
+
+    return handle;
+}
+
+OE_INLINE int _getaddrinfo_read(
```

Comment:
Why does this function need to be inline?

Reply:
It doesn't. Will remove.